scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala

```diff
@@ -107,19 +107,22 @@ class RedundantBraces(ftoks: FormatTokens) extends FormatTokensRewrite.Rule {
     def replace = replaceToken("(") {
       val rt = ft.right
       new Token.LeftParen(rt.input, rt.dialect, rt.start)
-    } // we will not keep it but will hint to onRight
-    ft.meta.rightOwner match {
-      case t: Term.Function if t.tokens.last.is[Token.RightBrace] =>
+    }
+    // we will not keep it but will hint to onRight
+    (ft.meta.rightOwner, ft.left) match {
+      case (_, _: Token.MacroSplice) =>
+        null
+      case (t: Term.Function, _) if t.tokens.last.is[Token.RightBrace]  =>
         if (!okToRemoveFunctionInApplyOrInit(t)) null
         else if (okToReplaceFunctionInSingleArgApply(t)) replace
         else removeToken
-      case t: Term.Block =>
+      case (t: Term.Block, _) =>
```

Comment:
You can just change one line here:
```suggestion
      case t: Term.Block if ft.left.isNot[Token.MacroSplice] =>
```

Reply:
i would even put it after `=>`:

```
  if (ft.left.is[...]) null
  else if ...
```

Reply:
Was not aware of `is` / `isNot`. I've used both suggestions to some extend but the end result is slightly different